roles/ceph-osd/templates/ceph-osd-run.sh.j2

```diff
@@ -24,21 +24,19 @@ function expose_devices {
 }
 
 function expose_partitions {
+  local tmp_disk
+  tmp_disk=$(mktemp)
+  docker run --name=expose_partitions --privileged=true -v /dev:/dev --entrypoint /usr/sbin/ceph-disk {{ ceph_docker_registry }}/{{ ceph_docker_image }}:{{ ceph_docker_image_tag }} list /dev/${1} > $tmp_disk
   local partition
-  for partition in Block.wal Block.db Journal; do
-    if docker logs ceph-osd-prepare-{{ ansible_hostname }}-devdev${1} |& grep -Eo "$partition is GPT partition"; then
-      if [[ "$partition" == "Block.wal" ]]; then
-        part=$(docker logs ceph-osd-prepare-{{ ansible_hostname }}-devdev${1} |& grep "$partition is GPT partition" | grep -Eo /dev/disk/by-partuuid/${REGEX} | uniq)
-        DOCKER_ENV="$DOCKER_ENV -e OSD_BLUESTORE_BLOCK_WAL=$part"
-      elif [[ "$partition" == "Block.db" ]]; then
-        part=$(docker logs ceph-osd-prepare-{{ ansible_hostname }}-devdev${1} |& grep "$partition is GPT partition" | grep -Eo /dev/disk/by-partuuid/${REGEX} | uniq)
-        DOCKER_ENV="$DOCKER_ENV -e OSD_BLUESTORE_BLOCK_DB=$part"
-      elif [[ "$partition" == "Journal" ]]; then
-        part=$(docker logs ceph-osd-prepare-{{ ansible_hostname }}-devdev${1} |& grep "$partition is GPT partition" | grep -Eo /dev/disk/by-partuuid/${REGEX} | uniq)
-        DOCKER_ENV="$DOCKER_ENV -e OSD_JOURNAL=$part"
-      fi
+  for partition in block.wal block.db journal; do
+    if grep -o $partition $tmp_disk; then
+      # NOTE(lesen): awk looks for a pattern and print the next column after it
+      journal=$(awk -v pattern=${partition} '$0 ~ pattern { f=NR }; f && (NR-1 == f)' RS=" " $tmp_disk)
```

Comment:
this looks brittle, do you have examples on how this really works with input from ceph-disk? I couldn't make it work nicely

    $ ceph-disk list sdb
    /dev/sdb :
     /dev/sdb1 ceph data, active, cluster ceph, osd.16, block /dev/sdb2
     /dev/sdb2 ceph block, for /dev/sdb1
    $ ceph-disk list sdb > listed
    $ awk -v pattern=block '$0 ~ pattern { f=NR }; f && (NR-1 == f)' RS=" " listed
    /dev/sdb2
    
    for

Reply:
```
[root@osd0 ~]# docker run --privileged=true -v /dev:/dev --entrypoint /usr/sbin/ceph-disk docker.io/ceph/daemon:tag-build-master-luminous-ubuntu-16.04 list /dev/sda > sda
[root@osd0 ~]# cat sda
/dev/sda :
 /dev/sda1 ceph data, prepared, unknown cluster 4ba83578-c524-4c4c-a5fc-be6c40b6bef1, osd.0, block /dev/sda2, block.db /dev/sdc1, block.wal /dev/sdc2
 /dev/sda2 ceph block, for /dev/sda1
[root@osd0 ~]# partition=block.db ; awk -v pattern=${partition} '$0 ~ pattern { f=NR }; f && (NR-1 == f)' RS=" " sda
/dev/sdc1,
[root@osd0 ~]# partition=block.db ; part=$(awk -v pattern=${partition} '$0 ~ pattern { f=NR }; f && (NR-1 == f)' RS=" " sda)
[root@osd0 ~]# echo ${part//,}
/dev/sdc1
```